wp-admin/js/widgets/media-widgets.js

```diff
@@ -713,6 +713,20 @@ wp.mediaWidgets = ( function( $ ) {
 				modelProps.attachment_id = mediaFrameProps.id;
 			}
 
+			isEmbed = 0 === modelProps.attachment_id;
+
+			// Reset all video/audio sources.
+			wp.media.view.settings.embedExts.map( function( ext ) {
+
+				// Only reset when the main source changes.
+				if ( ! ( ext in modelProps ) || isEmbed ) {
+					modelProps[ ext ] = '';
+				}
+			} );
+
+			extension = mediaFrameProps.url.split( '.' ).pop().toLowerCase();
+			modelProps[ extension ] = mediaFrameProps.url;
```

Comment:
If `extension` is not valid it will be purged by the instance schema. 

Do you want us to be more verbose here and test against an extension whitelist?